### ISSUE TITLE
[STORM-2887] Add RocksDB storage and Basic Metrics forwarding from the supervisor

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#6669]
<p>This portion is just to put in the rocksdb storage engine, the APIs to be able to store the metrics, and to update the supervisor to send the few metrics that it knows about.</p>

<p>Memory usage when using cgroups.</p>

<p>This is by no means complete but should be able to unblock some work on elasticity in the scheduler.</p>

---
<details><summary><i>Originally reported by <a title="Robert Joseph Evans" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=revans2">revans2</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-2887" target="_blank">Add RocksDB storage and Basic Metrics forwarding from the supervisor</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Aaron Gresch" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=agresch">agresch</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2018-01-22T21:24:31+00:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>

**Comment by jira-importer** [comment]
Subtask of parent task STORM-2156

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kabhwan">kabhwan</a>:</i>
<p>Thanks <a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=agresch" class="user-hover" rel="agresch">Aaron Gresch</a>, I merged into master.</p>
